pippy/count_flops.py

```diff
@@ -0,0 +1,160 @@
+import torch
+from torch.fx.passes.shape_prop import ShapeProp
+from torch.fx.passes.split_module import split_module
+from torch_mlir import run_pipeline_with_repro_report
+from torch_mlir.dialects.torch.importer.jit_ir import ClassAnnotator, ModuleBuilder
+
+# This unused import is necessary for linalg passes to be registered.
+# noinspection PyUnresolvedReferences
+from torch_mlir_e2e_test.linalg_on_tensors_backends.refbackend import RefBackendInvoker # pylint: disable=unused-import
+
+torch.fx.Tracer.proxy_buffer_attributes = True
+
+PIPELINE = [
+    "torchscript-module-to-torch-backend-pipeline",
+    "torch-backend-to-linalg-on-tensors-backend-pipeline",
+    "func.func(linalg-bufferize)",
+    "func.func(convert-linalg-to-affine-loops)",
+    # The workhorse of this script - unroll "affine loops" with trip count less than 1000000 (i.e. all of them)
+    # completely.
+    "func.func(affine-loop-unroll{ unroll-full unroll-full-threshold=1000000 })",
+]
+
+
+def script_module_with_annotations(test_module, annotations):
+    class_annotator = ClassAnnotator()
+    recursivescriptmodule = torch.jit.script(test_module)
+    frozen_recursivescriptmodule = recursivescriptmodule
+    class_annotator.exportNone(frozen_recursivescriptmodule._c._type())
+    class_annotator.exportPath(frozen_recursivescriptmodule._c._type(), ["forward"])
+    class_annotator.annotateArgs(
+        frozen_recursivescriptmodule._c._type(), ["forward"], annotations
+    )
+    return frozen_recursivescriptmodule, class_annotator
+
+
+def compile_ts_module_to_mlir_module(mod, in_shapes):
+    recursivescriptmodule, class_annotator = script_module_with_annotations(
+        mod, [None] + [(in_shape, torch.float32, True) for in_shape in in_shapes]
+    )
+
+    mb = ModuleBuilder()
+    mb.import_module(recursivescriptmodule._c, class_annotator)
+    run_pipeline_with_repro_report(mb.module, ",".join(PIPELINE), "")
+
+    return mb.module
+
+
+PARTITION_COUNTER = 0
+
+
+def mod_partition(_node):
+    global PARTITION_COUNTER
+    partition = PARTITION_COUNTER
+    PARTITION_COUNTER += 1
+    return partition
+
+
+def find_op_in_submodule(submodule):
+    for node in submodule.graph.nodes:
+        if node.op in {"call_function", "call_module"}:
+            return node.name
+    return None
+
+
+def compile_model_op_by_op(model, example_input):
+    """
+    High-level this 
```

Comment:
Maybe add a colon here so it's clear this is starting the numbered list? (I assume that's what this is here for)